app/src/main/java/uk/ac/bath/petmatch/MainActivity.java

```diff
@@ -186,12 +204,15 @@ public boolean onNavigationItemSelected(MenuItem item) {
                     ShelterProfileActivity.class);
             startActivity(startShelterProfileIntent);
 
-        } else if (id == R.id.nav_settings) {
+        } else if (id == R.id.nav_user_capabilities) {
 
-            // handles user capabilities
-            Intent startUserCapabilitiesIntent = new Intent(getApplicationContext(),
-                    UserCapabilitiesActivity.class);
-            startActivity(startUserCapabilitiesIntent);
+            if(loginService.isUserLoggedIn()) {
+                Intent startUserCapabilitiesIntent = new Intent(getApplicationContext(),
+                        UserCapabilitiesActivity.class);
+                startActivity(startUserCapabilitiesIntent);
+            } else {
+                displayNoLoginErrorAlert();
```

Comment:
I really like the message, but also put it inside the UserCapabilitiesActivity, if a user is accidentally directed there and is not logged in